rust/main/agents/relayer/src/msg/metadata/base_builder/validator_announced_storages.rs

```diff
@@ -0,0 +1,73 @@
+use eyre::Context;
+
+use hyperlane_base::cache::FunctionCallCache;
+use hyperlane_core::{ValidatorAnnounce, H256};
+
+const DOMAIN_NAME: &str = "";
+const METHOD_NAME: &str = "get_announced_storage_locations";
+
+/// Helper function to fetch storage locations for validators.
+pub async fn fetch_storage_locations_helper(
+    validators: &[H256],
+    cache: &impl FunctionCallCache,
+    validator_announce: &dyn ValidatorAnnounce,
+) -> eyre::Result<Vec<Vec<String>>> {
+    const CTX: &str = "When fetching storage locations";
+
+    let mut storage_locations = Vec::new();
+    let mut missing_validators = Vec::new();
+
+    for (index, validator) in validators.iter().enumerate() {
+        let key = generate_cache_key(validator);
+
+        // Attempt to retrieve from cache
+        if let Some(cached) = cache
+            .get_cached_call_result::<Vec<String>>(DOMAIN_NAME, METHOD_NAME, &key)
```

Comment:
the validator announce is specific to the origin domain (i.e. the domain that the `&dyn ValidatorAnnounce` relates to). I think as written with the `DOMAIN_NAME` being `""`, this would mean that if a validator address is used for multiple different chains (which is often the case, many operators use the same validator key for validating different chains) then we'd get wrong data in our cache hit

I think you can use `validator_announce.domain().name()` to get the domain name

Reply:
Added